xbmc/FileItemList.cpp

```diff
@@ -705,7 +705,31 @@ void CFileItemList::RemoveExtensions()
     m_items[i]->RemoveExtension();
 }
 
-void CFileItemList::Stack(bool stackFiles /* = true */)
+void CFileItemList::ConvertDiscFoldersToFiles()
+{
+  for (const auto& item : m_items)
+  {
+    const std::string playPath = VIDEO::UTILS::GetOpticalMediaPath(*item);
+    if (!playPath.empty())
+      ConvertDiscFolderToFile(item, playPath);
+  }
+}
+
+void CFileItemList::ConvertDiscFolderToFile(const std::shared_ptr<CFileItem>& item,
+                                            const std::string& playPath)
+{
+
+  item->SetPath(playPath); // updated path (for DVD/Bluray files)
+  item->m_bIsFolder = false;
+  item->SetLabel2("");
```

Comment:
Why are you resetting label2?

General thought: Instead of manipulating `item` should rather a new item be created? So, change `void CFileItemList::ConvertDiscFolderToFile` to `std::shared_ptr<CFileItem> CFileItemList::CreateFileForDiscFolder`?